lms/static/scripts/frontend_apps/components/BasicLTILaunchApp.js

```diff
@@ -219,37 +226,75 @@ export default function BasicLTILaunchApp() {
    * Report a submission to the LMS, with the LMS-provided metadata needed for
    * later grading of the assignment.
    */
-  const reportSubmission = useCallback(async () => {
-    // If a teacher launches an assignment or the LMS does not support reporting
-    // outcomes or grading is not enabled for the assignment, then no submission
-    // URL will be available.
-    if (!canvas.speedGrader || !canvas.speedGrader.submissionParams) {
-      return;
-    }
+  const reportSubmission = useCallback(
+    /** @param {string} [submittedAt] - ISO8601 date for the submission */
+    async submittedAt => {
+      // If a teacher launches an assignment or the LMS does not support reporting
+      // outcomes or grading is not enabled for the assignment, then no submission
+      // URL will be available.
+      if (!canvas.speedGrader || !canvas.speedGrader.submissionParams) {
+        return;
+      }
+
+      // Don't submit before content is viewable
+      if (!contentReady) {
+        return;
+      }
+
+      const submissionParams = {
+        ...canvas.speedGrader.submissionParams,
+        submitted_at: submittedAt ?? undefined,
+      };
 
-    // Don't submit before content is viewable
-    if (!contentReady) {
+      try {
+        await apiCall({
+          authToken,
+          path: '/api/lti/submissions',
+          data: submissionParams,
+        });
+      } catch (e) {
+        // If reporting the submission failed, replace the content with an error.
+        // This avoids the student trying to complete the assignment without
+        // knowing that there was a problem, and the teacher then not seeing a
+        // submission.
+        handleError(e, 'error-reporting-submission', false);
+      }
+    },
+    [authToken, canvas.speedGrader, contentReady]
+  );
+
+  // Submit an empty grading submission on behalf of this student user
+  // unless configured to wait until after annotation activity occurs.
+  useEffect(() => {
+    if (submitAfterActivity) {
       return;
     }
+    reportSubmission();
+  }, [reportSubmission, submitAfterActivity]);
 
-    try {
-      await apiCall({
-        authToken,
-        path: '/api/lti/submissions',
-        data: canvas.speedGrader.submissionParams,
-      });
-    } catch (e) {
-      // If reporting the submission failed, replace the content with an error.
-      // This avoids the student trying to complete the assignment without
-      // knowing that there was a problem, and the teacher then not seeing a
-      // submission.
-      handleError(e, 'error-reporting-submission', false);
+  // Respond to the first qualifying annotation activity from the client:
+  // submit an empty grading submission if configured to do so.
+  useEffect(() => {
+    const unsubscribe = () =>
+      clientRPC.off('annotationActivity', onAnnotationActivity);
+
+    /**
+     * @param {string} eventType
+     * @param {object} data
+     *   @param {string} [data.date]
```

Comment:
Hmmm, I guess I was nervous about assuming its presence because it comes from an outside system that we can't ensure the behavior of. It _should_ be present...I can go ahead and make this required if you think that seems right.